packages/contracts-bedrock/test/dispute/AnchorStateRegistry.t.sol

```diff
@@ -26,117 +30,588 @@ contract AnchorStateRegistry_Init is FaultDisputeGame_Init {
-        (Hash root, uint256 l2BlockNumber) = anchorStateRegistry.anchors(gameProxy.gameType());
-        assert(l2BlockNumber < gameProxy.l2BlockNumber());
+    /// @notice Tests that tryUpdateAnchorState will succeed if the game is valid, the game block
+    ///         number is greater than the current anchor root block number, and the game is the
+    ///         currently respected game type.
+    /// @param _l2BlockNumber The L2 block number to use for the game.
+    function testFuzz_tryUpdateAnchorState_validNewerState_succeeds(uint256 _l2BlockNumber) public {
+        // Grab block number of the existing anchor root.
+        (Hash root, uint256 l2BlockNumber) = anchorStateRegistry.getAnchorRoot();
+
+        // Bound the new block number.
+        _l2BlockNumber = bound(_l2BlockNumber, l2BlockNumber + 1, type(uint256).max);
+
+        // Mock the l2BlockNumber call.
+        vm.mockCall(address(gameProxy), abi.encodeCall(gameProxy.l2BlockNumber, ()), abi.encode(_l2BlockNumber));
 
-        // Mock the state that we want.
+        // Mock the DEFENDER_WINS state.
         vm.mockCall(address(gameProxy), abi.encodeCall(gameProxy.status, ()), abi.encode(GameStatus.DEFENDER_WINS));
 
-        // Try to update the anchor state.
+        // Make our game type the respected game type.
+        vm.mockCall(
+            address(optimismPortal2),
+            abi.encodeCall(optimismPortal2.respectedGameType, ()),
+            abi.encode(gameProxy.gameType())
+        );
+
+        // Update the anchor state.
         vm.prank(address(gameProxy));
+        vm.expectEmit(address(anchorStateRegistry));
+        emit AnchorUpdated(gameProxy);
         anchorStateRegistry.tryUpdateAnchorState();
 
         // Confirm that the anchor state is now the same as the game state.
-        (root, l2BlockNumber) = anchorStateRegistry.anchors(gameProxy.gameType());
+        (root, l2BlockNumber) = anchorStateRegistry.getAnchorRoot();
         assertEq(l2BlockNumber, gameProxy.l2BlockNumber());
         assertEq(root.raw(), gameProxy.rootClaim().raw());
+
+        // Confirm that the anchor game is now set.
+        IFaultDisputeGame anchorGame = anchorStateRegistry.anchorGame();
+        assertEq(address(anchorGame), address(gameProxy));
     }
+}
 
-    /// @dev Tests that updating the anchor state fails when the game state is valid but older.
-    function test_tryUpdateAnchorState_validOlderState_fails() public {
-        // Confirm that the anchor state is newer than the game state.
-        vm.mockCall(address(gameProxy), abi.encodeCall(gameProxy.l2BlockNumber, ()), abi.encode(0));
-        (Hash root, uint256 l2BlockNumber) = anchorStateRegistry.anchors(gameProxy.gameType());
-        assert(l2BlockNumber >= gameProxy.l2BlockNumber());
+contract AnchorStateRegistry_TryUpdateAnchorState_TestFail is AnchorStateRegistry_Init {
+    /// @notice Tests that tryUpdateAnchorState will revert if the game is valid and the game block
```

Comment:
here and elsewhere in this test contract, we see language like "will revert" and "TestFail." But we don't expect `tryUpdateAnchorState` to revert in any case, and perhaps these could be called `_succeeds`